src/ArborX_LinearBVH.hpp

```diff
@@ -247,43 +246,45 @@ BasicBoundingVolumeHierarchy<MemorySpace, BoundingVolume, Enable>::
                      Kokkos::MemoryUnmanaged>(&_bounds),
         Kokkos::View<BoundingVolume, MemorySpace, Kokkos::MemoryUnmanaged>(
             &getBoundingVolume(getRoot())));
-    Kokkos::Profiling::popRegion();
     return;
   }
 
-  Kokkos::Profiling::pushRegion("ArborX::BVH::BVH::assign_morton_codes");
-
-  // calculate Morton codes of all objects
-  Kokkos::View<unsigned int *, MemorySpace> morton_indices(
-      Kokkos::view_alloc(Kokkos::WithoutInitializing,
-                         "ArborX::BVH::BVH::morton"),
-      size());
-  Details::TreeConstruction::assignMortonCodes(space, primitives,
-                                               morton_indices, bbox);
-
-  Kokkos::Profiling::popRegion();
-  Kokkos::Profiling::pushRegion("ArborX::BVH::BVH::sort_morton_codes");
+  Kokkos::View<unsigned int *, MemorySpace> morton_indices;
+  { // calculate Morton codes of all objects
+    KokkosExt::ScopedProfileRegion guard_assign_morton_codes(
+        "ArborX::BVH::BVH::assign_morton_codes");
+
+    morton_indices = Kokkos::View<unsigned int *, MemorySpace>(
+        Kokkos::view_alloc(Kokkos::WithoutInitializing,
+                           "ArborX::BVH::BVH::morton"),
+        size());
+    Details::TreeConstruction::assignMortonCodes(space, primitives,
+                                                 morton_indices, bbox);
+  }
 
   // compute the ordering of primitives along Z-order space-filling curve
-  auto permutation_indices = Details::sortObjects(space, morton_indices);
+  auto permutation_indices = [&space, &morton_indices]() {
+    KokkosExt::ScopedProfileRegion guard_sort_morton_codes(
+        "ArborX::BVH::BVH::sort_morton_codes");
 
-  Kokkos::Profiling::popRegion();
-  Kokkos::Profiling::pushRegion("ArborX::BVH::BVH::generate_hierarchy");
+    return Details::sortObjects(space, morton_indices);
+  }();
```

Comment:
This immediately invoked lambda expression is there for illustration only.  I am not convinced this one is a good change for instance.  I have it as a basis for discussion.